### PR TITLE
docs: migrate to sphinx

### PR DESCRIPTION
**Pull request by classabbyamp**
# TODO

- [x] add manpages to sphinx
- [x] add online docs to sphinx
- [x] address TODOs left in the docs
- [x] readthedocs integration
- [x] logo? (the current one is kinda a placeholder)
- [ ] move/simplify README
- [ ] unify main website content and reference docs
- [ ] remove generate-zbm.8 from bin/generate-zbm (how should the help now be shown?)
- [ ] migrate manpage releng stuff
- [ ] migrate in-zbm online docs to rst2ansi (and rm /docs/pod/online)
    - https://github.com/zbm-dev/zfsbootmenu/compare/doc-sync-tooling
- [ ] internal function documentation

### CODE REVIEW

**Comment by zdykstra** [comment]
Oof. Thank you @classabbyamp. I really appreciate your time and efforts on this.  What are your thoughts on replacing the `:glob:` usage with an explicit ordering of guides?  It'd make it slightly less automatic to add a new guide, but updating table of contents shouldn't be too onerous.

I'll need to read up on how to do automatic docs deployments (which webhooks have to be enabled and where, etc), unless that all is more-or-less magic on the RTD side.

@ahesford do we want to add the online/embedded documentation to this? I don't mind if we do, as long as the rst2ansi output is sane and can be parsed by the help system.

**Comment by classabbyamp** [comment]
> What are your thoughts on replacing the :glob: usage with an explicit ordering of guides? It'd make it slightly less automatic to add a new guide, but updating table of contents shouldn't be too onerous.

fine by me

> I'll need to read up on how to do automatic docs deployments (which webhooks have to be enabled and where, etc), unless that all is more-or-less magic on the RTD side.

I believe that once we add the [config file](https://docs.readthedocs.io/en/stable/config-file/index.html) rtd should pick up all tags and pushes to certain branches automatically.

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  :maxdepth: 3
+  :titlesonly:
+
+  guides/general
+  guides/debian
+  guides/void-linux
+  guides/third-party
+
+.. toctree::
+  :caption: Online
```
Comment (classabbyamp) [inline docs/index.rst:32]: dunno what to title this

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  :maxdepth: 3
+  :titlesonly:
+
+  online/main-screen
+  online/diff-viewer
+  online/kernel-management
+  online/recovery-shell
+  online/snapshot-management
+  online/zpool-health
```
Comment (classabbyamp) [inline docs/index.rst:41]: is this order good?

**Pull request merged**